docs/src/pages/components/data-grid/demo/FullFeauredDemo.tsx

```diff
@@ -0,0 +1,153 @@
+import * as React from 'react';
+import { XGrid, GridToolbar } from '@material-ui/x-grid';
+import {
+  useDemoData,
+  getCommodityColumns,
+  getEmployeeColumns,
+  getRealData,
```

Comment:
How about we only use `useDemoData` (or lazy generate the new data, for later :p)?

```suggestion
```

`useDemoData` is optimized to generate data efficiently. Here is the current lag when picking the 100,000 page size option. It's over 5s (in production), it's not an option.

<img width="463" alt="Screenshot 2021-03-02 at 20 39 15" src="https://user-images.githubusercontent.com/3165635/109704850-603fbc80-7b97-11eb-94ef-f4eaed559aa8.png">

Damien has built a static .json, it loads fast enough, loading 1,000 rows. However, it's the same data duplicated x100 to reach 100,000 rows. It's doesn't yield a great sense of randomness in the dispalyed information:

<img width="538" alt="Screenshot 2021-03-02 at 20 42 44" src="https://user-images.githubusercontent.com/3165635/109705437-11deed80-7b98-11eb-89ea-b6bbdb591e05.png">

It would be interesting to see how ag-Grid solve this problem with their large demos

Reply:
So how about we solve the issue the following way -> remove the options bar at the top, keep the data set as "Commodity", make the rows be "more" by default, and set default pagination. That way we are kind of showing the most complete state of the grid. If someone wants to play more they can always open the codesandbox and change the settings (another benefit will be that the demo code will be shorter).

Reply:
I personally think that it's important we show:

- we can handle a **lot** of data, 100,000 or even 1,000,000 would be ideal. With 1,000,000 we will definitely need to lazy load them. It might also hit the max scrollbar height limit.
- we support multiple column types. Having multiple data set can help achieve this goal, but not strong point of view.
- we support the most common use cases, like trading commodities or employes data sets, so two type of data sets makes sens

Reply:
Ok how about this: we keep it as it is now but introduce a loader (it will also showcase the grid loader feature) and I'll add a theme switcher (I think there is one in the demo app)?

Reply:
There is already a dark/light theme switch. I don't think that the demo should have one.

For the loading of data, I would personally rather have 10,000 rows than showing a 5s complet freeze of the JavaScript thread. ag-grid takes a while to load the rows but it's not freezing which is a major difference.

Now, I don't understand why only using `useDemoData` only isn't simpler. For showijg a spinner, we can easily add a hook that defers updates

Reply:
I solved the freeze, it is way more responsive now (only using the `useDemoData` + its API)